### DISCUSSION TITLE
Java 17 Compatibility with csrfguard-4.1.4

### DISCUSSION THREAD

**Discussion by pcwd111** [OWASP/www-project-csrfguard#196] [Q&A] [answered]
Is csrfguard-4.1.4.jar compatible with Java 17?

**Comment by forgedhallpass** [comment, marked as answer]
There is a related [issue](https://github.com/OWASP/www-project-csrfguard/issues/179) to Java 17, which I have fixed, but it's not yet released. Feel free to build the latest version of the `master` branch, test it out and let me/us know if you are facing any additional issues.

**Reply by forgedhallpass** [reply]
Released under version [4.3.0](https://github.com/OWASP/www-project-csrfguard/releases/tag/4.3.0)

**Comment by bvjagan** [comment]
Hi there, 
even with version 4.3.0 its not working with Java 17 with Tomcat 10.1 as the CsrfGuardHttpSessionListener implements javax.servlet.http.HttpSessionListener instead of jakarta.servlet.http.HttpSessionListener.
Am i missing something here? Appreciate for any input here. 
Thank you.

**Reply by forgedhallpass** [reply]
Hello @bvjagan,

Have you seen: https://github.com/OWASP/www-project-csrfguard/releases/tag/4.3.0-jakarta
Issue reference: https://github.com/OWASP/www-project-csrfguard/issues/176
